R/sdf_utils.R

```diff
@@ -0,0 +1,32 @@
+#' Fast cbind for Spark DataFrames
+#'
+#' This is a version of `sdf_bind_cols` that works by zipping
+#' RDDs. From the API docs: "Assumes that the two RDDs have the
+#' *same number of partitions* and the *same number of elements
+#' in each partition* (e.g. one was made through a map on the
+#' other)."
+#'
+#' @keywords internal
+#'
+#' @param ... Spark DataFrames to cbind
+sdf_fast_bind_cols <- function(...) {
```

Comment:
Checking the number of partitions is easy. Counting elements in each partition is more expensive since we have to iterate through them (albeit in parallel.) Maybe we can have a `validate` argument in the function that defaults to `TRUE`?